### DISCUSSION TITLE
How to add nullable return type declaration?

### DISCUSSION THREAD

**Discussion by brandonkelly** [rectorphp/rector#6665] [Q&A] [answered]
I need to configure an `AddReturnTypeDeclarationRector` rule similar to [this example](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#addreturntypedeclarationrector), except the return type must be nullable. For example:

```diff
 class SomeClass
 {
-    public function getData()
+    public function getData(): ?array
     {
     }
 }
```

How exactly can that be expressed as a `PHPStan\Type\Type` object for `AddReturnTypeDeclarationRector`?

**Comment by samsonasik** [comment, marked as answer]
You possibly can do with:

```php
use PHPStan\Type\UnionType;
use PHPStan\Type\ArrayType;
use PHPStan\Type\NullType;
use PHPStan\Type\MixedType;

new UnionType([new ArrayType(new MixedType(), new MixedType()), new NullType()])
```
